File: Data Structures & Algorithms/count-connected-components/submission-1.py
class Solution:
    def countComponents(self, n: int, edges: List[List[int]]) -> int:
        # Construction du graphe (Adjacence)
        g = {i: [] for i in range(n)}
        for u, v in edges:
            g[u].append(v)
            g[v].append(u)

        visited = set()

        def dfs(node):
            visited.add(node)

            for nei in g[node]:
                if nei not in visited:
                    dfs(nei)
        
        count = 0
        for i in range(n):
            if i not in visited:
                dfs(i)
                count += 1
        return count 
        

File: Data Structures & Algorithms/longest-palindromic-substring/submission-1.cpp
class Solution {
public:
    int maxC = 0;
    string longestP = "";
    
    void dfs(string s, int l, int r) {
        int count = 0;
        while (l >= 0 && r < s.size() && s[l] == s[r]) {
            count++;
            l--;
            r++;
        }

        int length = r - l - 1;
        if (length> longestP.size()) {
            maxC= count;
            longestP = s.substr(l+1, length);
        }
    }
    
    string longestPalindrome(string s) {
        for (int i = 0; i < s.size(); i++) {
            dfs(s, i, i);
            dfs(s, i, i + 1);
        }

        return longestP;
    }
};
